core/src/main/scala/dev/profunktor/auth/jwt.scala

```diff
@@ -25,17 +27,52 @@ object jwt {
 
   case class JwtToken(value: String) extends AnyVal
 
-  case class JwtSecretKey(value: String) extends AnyVal
+  object JwtSecretKey {
+    def apply(key: Array[Byte]): JwtSecretKey = new JwtSecretKeyByteArr(key)
+    def apply(key: Array[Char]): JwtSecretKey = new JwtSecretKeyCharArr(key)
+    def apply(key: PrivateKey): JwtSecretKey  = new JwtSecretKeyPK(key)
+  }
+  sealed trait JwtSecretKey {
```

Comment:
As we convert every type of key to `Array[Byte]` anyways: maybe it would be cleaner to model `JwtSecretKey` as `case class JwtSecretKey(key: Array[Char])` and do the conversions in the apply methods?

Reply:
So some of the libs upon which you depend also have this same password as a String problem.  The way I wrote this PR, it delays making the password into a string as long as possible and avoiding that when the libs are fixed.  Pushing the conversion code into the apply method forces the conversions.  I'm trying to delay and avoid the conversions if possible.  Consider some new impl that just wants the raw bytes, forcing conversion early locks this out by design.

Reply:
That makes sense, thanks for explaining!